tests/tests_main/test_dependents.py

```diff
@@ -0,0 +1,63 @@
+from collections import defaultdict
+
+import pytest
+
+from pg_to_evalscript.process_graph_utils import get_execution_order
+from tests.utils import get_process_graph_json
```

Comment:
> Can we also add the test for a process graph where the same process would be used in different processes? For example the `rename_labels1` would get the values for the `target` argument from the `array_filter` process which is already used in the argument for `apply_dimension`.
> 
> Something like pg below, although this probably isn't correct because it doesn't work :
> 
> ```json
> {
>   "loadco1": {
>     "process_id": "load_collection",
>     "arguments": {
>       "id": "S2L1C",
>       "spatial_extent": {
>         "west": 16.1,
>         "east": 16.6,
>         "north": 48.6,
>         "south": 47.2
>       },
>       "temporal_extent": [
>         "2022-01-01",
>         "2022-02-01"
>       ],
>       "bands": [
>         "B01", "B02"
>       ]
>     }
>   },
>   "filter_every_second": {
>     "process_id": "apply_dimension",
>     "arguments": {
>       "data": {
>         "from_node": "loadco1"
>       },
>       "process": {
>         "process_graph": {
>           "filter": {
>             "process_id": "array_filter",
>             "arguments": {
>               "data": {
>                 "from_parameter": "data"
>               },
>               "condition": {
>                 "process_graph": {
>                   "mod2": {
>                     "process_id": "mod",
>                     "arguments": {
>                       "x": {
>                         "from_parameter": "index"
>                       },
>                       "y": 2
>                     }
>                   },
>                   "iseven": {
>                     "process_id": "eq",
>                     "arguments": {
>                       "x": {
>                         "from_node": "mod2"
>                       },
>                       "y": 0
>                     },
>                     "result": true
>                   }
>                 }
>               }
>             },
>             "result": true
>           }
>         }
>       },
>       "dimension": "t"
>     }
>   },
>   "filter_labels": {
>     "process_id": "array_filter",
>     "arguments": {
>       "data": {
>         "from_node": "loadco1"
>       },
>       "condition": {
>         "process_graph": {
>           "mod2": {
>             "process_id": "mod",
>             "arguments": {
>               "x": {
>                 "from_parameter": "index"
>               },
>               "y": 2
>             }
>           },
>           "iseven": {
>             "process_id": "eq",
>             "arguments": {
>               "x": {
>                 "from_node": "mod2"
>               },
>               "y": 0
>             },
>             "result": true
>           }
>         }
>       }
>     }
>   },
>   "rename_labels1": {
>     "process_id": "rename_labels",
>     "arguments": {
>       "data": {
>         "from_node": "filter_every_second"
>       },
>       "dimension": "t",
>       "target": {
>         "from_node": "filter_labels",
>       }
>     }
>   },
>   "resample": {
>     "process_id": "resample_cube_temporal",
>     "arguments": {
>       "data": {
>         "from_node": "loadco1"
>       },
>       "target": {
>         "from_node": "rename_labels1"
>       },
>       "valid_within": 4
>     }
>   },
>   "saveresult1": {
>     "process_id": "save_result",
>     "arguments": {
>       "data": {
>         "from_node": "resample"
>       },
>       "format": "png"
>     },
>     "result": true
>   }
> }
> ```

Yes, the process graph doesn't work because `filter_labels` node uses `array_filter` process , which expects an array as input, but we give it the `DataCube`  from `loadco1`.

I added the test for this, although I'm not sure what is intended to be achieved with 
> would get the values for the target argument from the array_filter process which is already used in the argument for apply_dimension.

When calculating the execution order, the processes used in nodes aren't relevant. We just check all the nodes, and what nodes they take in as inputs, and then we construct the order in which they can be executed, so that a node is only executed when all the nodes it takes as input have already been executed.